File: cs/numberTest.cs
using NUnit.Framework;

[TestFixture]
public class NumberTest
{

    [Test]
    public void Plus()
    {
        Assert.AreEqual(5, 3 + 2);
    }

    [Test]
    public void Minus()
    {
        Assert.AreEqual(1, 3 - 2);
    }

    [Test]
    public void Divide()
    {
        Assert.AreEqual(1, 3 / 2);
    }

    [Test]
    public void Multiply()
    {
        Assert.AreEqual(6, 3 * 2);
    }

    [Test]
    public void Remainder()
    {
        Assert.AreEqual(1, 3 % 2);
    }

    [Test]
    public void Power()
    {
        Assert.AreEqual(9, System.Math.Pow(3, 2));
    }

    [Test]
    public void StringToNumber()
    {
        Assert.AreEqual(1, System.Convert.ToDouble("1"));
        double actual;
        Assert.AreEqual(false, System.Double.TryParse("string", out actual));
        Assert.AreEqual(0, actual);
    }

    [Test]
    public void Abs()
    {
        Assert.AreEqual(10, System.Math.Abs(-10));
    }

    [Test]
    public void Floor()
    {
        Assert.AreEqual(1, System.Math.Floor(1.3));
    }

    [Test]
    public void Round()
    {
        Assert.AreEqual(2, System.Math.Round(1.5));
    }

    [Test]
    public void IEEE754Float()
    {
        Assert.AreEqual(110.00000000000001, 100 * 1.1);
        Assert.AreEqual(0.30000000000000004, 0.1 + 0.2);
    }

    [Test]
    public void AccurateFloatCalculation()
    {
        Assert.AreEqual(110, 100 * (decimal)1.1);
        Assert.AreEqual(0.3, 0.1m + 0.2m);
    }

    [Test]
    public void Cos()
    {
        // https://msdn.microsoft.com/en-us/library/system.math(v=vs.110).aspx
        Assert.AreEqual(-1, System.Math.Cos(System.Math.PI));
    }

}

File: nodejs/number_test.js
var assert = require('assert');

(function plus() {
  assert.equal(5, 3 + 2);
})();

(function minus() {
  assert.equal(1, 3 - 2);
})();

(function divide() {
  assert.equal(1.5, 3 / 2);
})();

(function multiply() {
  assert.equal(6, 3 * 2);
})();

(function remainder() {
  assert.equal(1, 3 % 2);
})();

(function power() {
  assert.equal(9, Math.pow(3, 2));
})();


(function stringToNumber() {
  assert.equal(1, parseInt("1"));
  assert.equal(true, isNaN(parseInt(null)));
  assert.equal(1.1, parseFloat("1.1"));
})();

(function abs() {
  assert.equal(10, Math.abs(-10));
})();

(function floor() {
  assert.equal(1, Math.floor(1.3));
})();

(function round() {
  assert.equal(2, Math.round(1.5));
  assert.equal(1.5, Math.round(1.54 * 10) / 10);
})();


(function ieee754Float() {
  assert.equal(110.00000000000001, 100 * 1.1);
  assert.equal(0.30000000000000004, 0.1 + 0.2);
})();

(function accurateFloatCalculation() {
  // No support
})();

(function cos() {
  // http://www.w3schools.com/jsref/jsref_obj_math.asp
  assert.equal(-1, Math.cos(Math.PI));
})();
